destination/writer/writer.go

```diff
@@ -0,0 +1,302 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/doug-martin/goqu/v9"
+
+	"github.com/conduitio-labs/conduit-connector-vitess/columntypes"
+
+	// we need the import to work with the mysql dialect.
+	_ "github.com/doug-martin/goqu/v9/dialect/mysql"
+)
+
+const (
+	// metadata related.
+	metadataTable = "vitess.table"
+)
+
+// Writer implements a writer logic for Vitess destination.
+type Writer struct {
+	db          *sql.DB
+	table       string
+	keyColumn   string
+	columnTypes map[string]string
+}
+
+// Params is an incoming params for the NewWriter function.
+type Params struct {
+	DB        *sql.DB
+	Table     string
+	KeyColumn string
+}
+
+// NewWriter creates new instance of the Writer.
+func NewWriter(ctx context.Context, params Params) (*Writer, error) {
+	writer := &Writer{
+		db:        params.DB,
+		table:     params.Table,
+		keyColumn: params.KeyColumn,
+	}
+
+	columnTypes, err := columntypes.GetColumnTypes(ctx, writer.db, writer.table)
+	if err != nil {
+		return nil, fmt.Errorf("get column types: %w", err)
+	}
+	writer.columnTypes = columnTypes
+
+	return writer, nil
+}
+
+// InsertRecord inserts a sdk.Record into a Destination.
+func (w *Writer) InsertRecord(ctx context.Context, record sdk.Record) error {
```

Comment:
this method can also delete a record, which makes the name a bit confusing.

Reply:
Yeah, this is a good point. Thank you! I've renamed this method to `Write`.